### PR TITLE
MONGOID-5227 Add test and adjust docs for dotted attribute assignment in default scope

### PR DESCRIPTION
**Pull request by Neilshweky**

### CODE REVIEW

File: docs/reference/queries.txt [hunk 1]
```diff
@@ -1900,23 +1898,9 @@ not possible:
     default_scope ->{ where('tags.foo' => 'bar') }
```
Comment (p-mongo) [inline docs/reference/queries.txt:1898]: This should have an example showing what the result of `Band.new` is, would it have the `tags.foo` as the attribute name?

File: docs/reference/queries.txt [hunk 2]
```diff
@@ -1886,9 +1886,7 @@ in the default scope, the value in the default scope takes precedence:
 
   Band.new # => #<Band _id: 5c3f74ddce4ef3791abbb088, name: nil, active: false>
 
-Because a default scope initializes fields in new models as just described,
-defining a default scope with a dotted key and a simple literal value is
-not possible:
+As of Mongoid 8, a default scope can be defined with a dotted key:
```
Comment (johnnyshields) [inline docs/reference/queries.txt:1889]: When I first read this, I thought it was referring to dot/dollar key names. I'd recommend to be more explicit like the following:

```
As of Mongoid 8, a default scope can be defined using dot-notation for path traversal:
```

File: docs/reference/queries.txt (hunk 1 above)
Comment (Neilshweky) [reply docs/reference/queries.txt:1900]: I did not realize this would be a thing... I added docs and tests that demonstrate this.

**Review by p-mongo — APPROVED** [review]
Much appreciated.

File: docs/reference/queries.txt [hunk 3]
```diff
@@ -1887,8 +1887,8 @@ in the default scope, the value in the default scope takes precedence:
   Band.new # => #<Band _id: 5c3f74ddce4ef3791abbb088, name: nil, active: false>
 
 Because a default scope initializes fields in new models as just described,
-defining a default scope with a dotted key and a simple literal value is
-not possible:
+defining a default scope with a dotted key and a simple literal value, while
+possible, is not recommended.
```
Comment (p-mongo) [inline docs/reference/queries.txt:1891]:
```suggestion
possible, is not recommended:
```

**Pull request merged**